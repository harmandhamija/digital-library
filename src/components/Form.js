// Form.js
import { useState } from 'react';

const Form = (props) => {

    const [ input, setInput ] = useState('');

    const handleInput = (event) => {
        setInput(event.target.value);
    }

    const handleSubmit = (event) => {
        event.preventDefault();
        props.getUserInput(input);
        setInput('');
        props.setIsLoading(true);
    }

    const handleClickAlpha = (event) => {
        props.setSortYear(false)
        props.setSortAlpha(true)
    }

    const handleClickYear = (event) => {
        props.setSortAlpha(false)
        props.setSortYear(true)
    }

    return (
        <section className="form-section">
            <h1>Welcome to the digital book library!</h1>
            <div className="form-container">
                <form onSubmit={handleSubmit}>
                    <label htmlFor="Enter book title" className="sr-only">search for a book</label>
                    <input type="text" name="Enter book title" id="Enter book title" value ={input} placeholder="search for a book" onChange={handleInput} required></input>
                    <button type="submit">Search</button>
                </form>
                
                <div className="sort-buttons">
                    <button onClick={handleClickYear}>sort by year</button>
                    <button onClick={handleClickAlpha}>sort alphabetically</button>
                </div>
            </div>
        </section>
    )
}

export default Form;