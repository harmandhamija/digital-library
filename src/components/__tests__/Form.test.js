import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Form from '../Form';

afterEach(() => {
    cleanup();
});

test('should render form component', () => {
    render(<Form />);
    const formElement = screen.getByTestId('form');
    expect(formElement).toBeInTheDocument();
    expect(formElement).toHaveTextContent('search for a book');
})

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Form />, div);
})

// snapshot testing
it('matches snapshot', () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
})

