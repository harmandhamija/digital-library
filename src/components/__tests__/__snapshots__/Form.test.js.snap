// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="form-section"
  data-testid="form"
>
  <h1>
    Welcome to the digital book library!
  </h1>
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="sr-only"
        htmlFor="Enter book title"
      >
        search for a book
      </label>
      <input
        id="Enter book title"
        name="Enter book title"
        onChange={[Function]}
        placeholder="search for a book"
        required={true}
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      className="sort-buttons"
    >
      <button
        onClick={[Function]}
      >
        sort by year
      </button>
      <button
        onClick={[Function]}
      >
        sort alphabetically
      </button>
    </div>
  </div>
</section>
`;
